packages/belos/tpetra/example/PCPG/PCPGTpetraExFile.cpp

```diff
@@ -0,0 +1,437 @@
+        mass->insertGlobalValues(node, tuple(pos), tuple(m1));
+      }
+
+      if (iX > 0) {
+        pos = iX - 1 + iY * (numElePerDirection - 1);
+        stiff->insertGlobalValues(node, tuple(pos), tuple(k1));  // West
+        mass->insertGlobalValues(node, tuple(pos), tuple(m1));
+        if (iY > 0) {
+          pos = iX - 1 + (iY - 1) * (numElePerDirection - 1);
+          stiff->insertGlobalValues(node, tuple(pos), tuple(k1));  // North West
+          mass->insertGlobalValues(node, tuple(pos), tuple(m2));
+        }
+        if (iY < numElePerDirection - 2) {
+          pos = iX - 1 + (iY + 1) * (numElePerDirection - 1);
+          stiff->insertGlobalValues(node, tuple(pos), tuple(k1));  // South West
+          mass->insertGlobalValues(node, tuple(pos), tuple(m2));
+        }
+      }
+
+      if (iX < numElePerDirection - 2) {
+        pos = iX + 1 + iY * (numElePerDirection - 1);
+        stiff->insertGlobalValues(node, tuple(pos), tuple(k1));  // East
+        mass->insertGlobalValues(node, tuple(pos), tuple(m1));
+        if (iY > 0) {
+          pos = iX + 1 + (iY - 1) * (numElePerDirection - 1);
+          stiff->insertGlobalValues(node, tuple(pos), tuple(k1));  // North East
+          mass->insertGlobalValues(node, tuple(pos), tuple(m2));
+        }
+        if (iY < numElePerDirection - 2) {
+          pos = iX + 1 + (iY + 1) * (numElePerDirection - 1);
+          stiff->insertGlobalValues(node, tuple(pos), tuple(k1));  // South East
+          mass->insertGlobalValues(node, tuple(pos), tuple(m2));
+        }
+      }
+    }
+
+    stiff->fillComplete();
+    mass->fillComplete();
+
+    const ST one = SCT::one();
+    ST hdt = .00005;  // half time step
+
+    // A = Mass+Stiff*dt/2
+    RCP<MAT> A = Tpetra::MatrixMatrix::add(one, false, *mass, hdt, false, *stiff);
+
+    // B = Mass-Stiff*dt/2
+    hdt = -hdt;
+    RCP<MAT> B = Tpetra::MatrixMatrix::add(one, false, *mass, hdt, false, *stiff);
+
+    B->apply(*vecLHS, *vecRHS);
+
+    procVerbose = verbose && (comm->getRank() == 0);  // Only print on the zero processor
+
+    LHS = Teuchos::rcp_implicit_cast<MV>(vecLHS);
+    RHS = Teuchos::rcp_implicit_cast<MV>(vecRHS);
+
+    //
+    // ************Construct preconditioner*************
+    //
+    // Teuchos::ParameterList MLList; // Set MLList for Smoothed Aggregation
+
+    // ML_Tpetra::SetDefaults("SA", MLList); // reset parameters ML User's Guide
+    // MLList.set("smoother: type","Chebyshev"); // Chebyshev smoother  ...
+    // aztec?? MLList.set("smoother: sweeps",3); MLList.set("smoother: pre or
+    // post", "both"); // both pre- and post-smoothing
+
+    //
+    // #ifdef HAVE_ML_AMESOS
+    //     MLList.set("coarse: type","Amesos-KLU"); // solve with serial direct
+    //     solver KLU
+    // #else
+    //     MLList.set("coarse: type","Jacobi");     // not recommended
+    //     puts("Warning: Iterative coarse grid solve");
+    // #endif
```

Comment:
All this part of the code is commented. Maybe you can delete it?

Reply:
As for the PCPG test I think we have to keep it. Currently this part does not work but is part of the epetra version.
It might be useful later.